GLOSSARY.md

```diff
@@ -0,0 +1,299 @@
+handler (defined below) to `.then`. Unwrapping via `.then` creates a new
+`Promise` that will resolve to the return value of the handler. One promise may
+have zero-to-many child promises.
+
+Errors will propagate from a source promise to all child promises. If a
+rejected promise has no children, it is considered an unhandled rejection. It
+may be handled in a subsequent operation. If a promise has an error handler,
+propagation from source promises will halt at that promise. If a promise has
+many children, and only one handles errors, the other children will be
+unhandled rejections.
+
+Unwrapping `Promise<Promise<T>>` will automatically unwrap to `T`. That is:
+`.then(() => Promise.resolve(3)).then(val => console.log(val))` will
+output `3`.
+
+Internally, a promise is represented by a state machine that may be in one of
+three states — pending, fulfilled, or rejected. The fulfilled and rejected
+states are collectively referred to as "settled." Once a promise is settled, it
+cannot transition into any other state.
+
+### Executor / Resolver
+
+The function passed to the `new Promise` constructor. Run **synchronously**.
+Not in common direct use by most promise-users — used mainly to wrap sources of
+asynchrony. As such, it rarely resolves or rejects on the same tick. This API
+is provided for the express purpose of wrapping callback APIs.
+
+Catches all errors thrown. If an error is caught, the returned promise will be
+**rejected**. Because it starts life as a rejected promise with no children, it
+will immediately hit the `OnUnhandledRejection` callback.
+
+```js
+new Promise(function Excecutor (resolve, reject) {
+  // run immediately.
+})
+```
+
+### Handler
+
+A function passed to `Promise#then` or `Promise#catch`. If the function is
+passed as the first argument to `Promise#then`, it is a **fulfillment**
+handler. Otherwise, if it is passed to `.catch` or as the second argument to
+`.then`, it is a **rejection** handler. Fulfillment handlers are called with
+the resolved value of the parent promise. Rejection handlers are called with
+the error value of the parent promise. Calling `.then` or `.catch` on a Promise
+`A` returns a new `Promise`, which is a child of `A`. A handler which returns a
+value `T` or `Promise<T>` will resolve the new promise with `T`. A handler
+which throws an error `R`, or returns a `Promise` which itself eventually
+rejects (or has already rejected) with `R`, will reject the new promise with
+`R`.
+
+```js
+
+const p = new Promise(SomeOmittedExecutor)
+
+const p2 = p.then(function SuccessHandler(value) {
+  // fired if p fulfills,
+}, function ErrorHandler(err) {
+  // fired if p rejects.
+})
+
+const p3 = p2.catch(function ErrorHandler(value) {
+  // fired if p2 rejects.
+})
+
+const p4 = p3.then(function SuccessHandler (value) { 
+  // resolved with p2's value if p2 resolves,
+  // resolved with p3's value if p2 rejects and p3 resolves
+})
+```
+
+### Resolution / Settling
```

Comment:
See above, please distinguish resolution from settling. A promise can be deferred by resolving it to another pending promise, as is the case when you return a promise in a handler, call resolve() with a pending promise in an executor, or call Promise.resolve() with a pending promise explicitly. Settling only applies to the terminal resolution.